Document SDS for ingress gateways (#11164)
hashicorp/consul: website/content/docs/connect/config-entries/service-router.mdx

      description:
        'A list of HTTP response status codes that are eligible for retry.',
    },
    {
      yaml: false,
      name: 'RequestHeaders',
      type: 'HTTPHeaderModifiers: <optional>',
      description: `A set of [HTTP-specific header modification rules](/docs/connect/config-entries/service-router#httpheadermodifiers)
      that will be applied to requests routed to this service.
      This cannot be used with a \`tcp\` listener.`,
    },
    {
      yaml: false,
      name: 'ResponseHeaders',
      type: 'HTTPHeaderModifiers: <optional>',
      description: `A set of [HTTP-specific header modification rules](/docs/connect/config-entries/service-router#httpheadermodifiers)
      that will be applied to responses from this service.
      This cannot be used with a \`tcp\` listener.`,
    },
  ]}
/>

### `HTTPHeaderModifiers`

<ConfigEntryReference
  topLevel={false}
  yaml={false}
  keys={[
    {
      hcl: false,
      name: 'Unsupported',
      type: '',
      description: `HTTP Header modification is not yet supported in our Kubernetes CRDs.`,
    },
    {
      yaml: false,
      name: 'Add',
      type: 'map<string|string>: optional',

>       description: `The set of header value keyed by header name to
>       add. If a header with the same (case-insensitive) name already
>       exists, the value set here will be appended and both will be present.
>       If Envoy is used as the proxy, the value may contain

Reviewer: ```suggestion
      description: `The set of header names to add. If one or more header name values with the same (case-insensitive) name already exist, then the value set here will replace them all.
      If Envoy is used as the proxy, the value may contain
```
Same as below, does this make sense?

      [variable placeholders](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/headers#custom-request-response-headers) for example
      \`%DOWNSTREAM_REMOTE_ADDRESS%\` to interpolate dynamic request
      metadata into the value added.`,
    },
    {
      yaml: false,
      name: 'Set',
      type: 'map<string|string>: optional',

>       description: `The set of header value keyed by header name to
>       add. If one or more header values with the same (case-insensitive) name already exist,
>       the value set here will replace them all.
>       If Envoy is used as the proxy, the value may contain

Reviewer: ```suggestion
      description: `The set of header names to add. If one or more header name values with the same (case-insensitive) name already exist, then the value set here will replace them all.
      If Envoy is used as the proxy, the value may contain
```

@banks does this suggestion still convey what you were aiming for? The original sentence was a bit confusing to read.
